lib/queuePopulator/QueuePopulator.js

```diff
@@ -172,6 +152,35 @@ class QueuePopulator {
         });
     }
 
+    _loadExtensions(done) {
+        this._extensions = [];
+        async.each(Object.keys(this.extConfigs), (extName, done) => {
+            const extConfig = this.extConfigs[extName];
+            const index = require(`../../extensions/${extName}/index.js`);
+            // eslint-disable-next-line new-cap
+            const ext = new index.queuePopulatorExtension({
+                config: extConfig,
+                logger: this.log,
+            });
+            ext.setup(err => {
+                if (!err) {
+                    this.log.info(`${index.name} extension is ready`);
+                    this._extensions.push(ext);
+                } else {
+                    this.log.error('an error occurred loading ' +
+                                   `${index.name} extension`,
+                                   { error: err });
+                }
+                done(err);
+            });
+        }, done);
+    }
+
+    _unloadExtensions(done) {
+        async.each(this._extensions, (ext, done) => ext.shutdown(done),
+                   done);
```

Comment:
I originally did this to prepare integration of some setup/teardown from clueso. After some thought we decided not to put this code in the queue populator extensions but as other standalone tasks, and keep the extension focused to record filtering jobs.

I though it would not hurt to keep these methods as part of the extensions API, "just in case", but it may be better to remove them since the queue populator extensions should remain simple synchronous filters I doubt we will ever need async setup/teardown.

Reply:
To answer your question more directly, yes `this._extensions` is an array of references to extensions as class instances, which are valid and can be accessed also for asynchronous processing.

Reply:
cool! Thank you for the explanation :)